src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+`dart pub publish --force`.
+
+{{site.alert.note}}
+  At this point in this guide, anyone with push access to your
+  repository can publish new versions of the package.
+  Consider using [tag protection rules][sec-gh-tag-protection] or
+  [GitHub deployment Environments][sec-gh-environment] to limit who can
+  publish.
+{{site.alert.end}}
+
+[sec-gh-tag-protection]: #hardening-security-with-tag-protection-rules-on-github
+[sec-gh-environment]: #hardening-security-with-github-deployment-environments
+
+
+## Trigger automated publishing using Github Actions
+
+After you have completed the preceding activities, you can publish a
+new version of your package.
+
+1. Verify your package's version.
+
+    ```console
+    $ cat pubspec.yaml
+    package: my_package_name
+    version: 1.2.3            # must match the version number used in the git tag
+    environment:
+      sdk: ^2.18.0
+    ...
+    ```
+
+1. Tag the latest commit with the matching `tag-pattern`.
+
+    ```console
+    $ git tag v1.2.3          # assuming my tag pattern is: 'v{{version}}'
+    ```
+
+1. Push the tag to GitHub.
+
+    ```console
+    $ git push origin v1.2.3  # triggers the action that publishes my package.
+    ```
+
+1. Review the GitHub Action workflow logs at
+   `https://github.com/<organization>/<repository>/actions`.
+
+If you don't like using the `git` command to create tags, you can
+create _releases_ on GitHub.
+To learn more, check out [managing releases in a repository][4] in the
+GitHub documentation.
+
+
+## Troubleshoot issues with GitHub Actions
+
+* If the Action didn't trigger, check that the pattern configured in
+  `.github/workflows/publish.yml` matches the pushed _git tag_.
+* If the Action failed, the GitHub Action logs may contain clues as to
+  why it failed. If the log offered no clues, check that you set the
+  proper `organization/repository` and `tag-pattern` on `pub.dev`.
+
+Once published, you can see the publication event in the **Activity Log** on
+`pub.dev`.
+The **Activity Log** entry should contain a link to the GitHub Action run
+that published the package version.
+
+![Audit log after publishing from GitHub Actions](audit-log-pub-gh.png)
+
+## Hardening security of automated publishing
+
+### Hardening security with tag protection rules on GitHub
+
+
+Configuring automated publishing from GitHub Actions has one security
+flaw. Any GitHub user who can push a tag to your repository can publish
+your package.
+To restrict who can push tags to your repository, use
+[tag protection rules][5] on GitHub.
+
+By limiting who creates tags matching your `tag-pattern`,
+you limit who can publish the package.
+
+At this time, these [tag protection rules][5] lack flexibility. You may
+want to restrict who can trigger publishing using _GitHub Deployment
+environments_. The next section outlines this process.
+
+
+### Hardening security with GitHub Deployment Environments
+
+When configuring automated publishing from GitHub Actions in `pub.dev`,
```

Comment:
```suggestion
When configuring automated publishing from GitHub Actions on pub.dev,
```

`<code>` is not a emphasis, it's to highlight identifiers or code snippets to be copied directly.